uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/document.js

```diff
@@ -78,67 +85,74 @@ define([
 
           return deferred.promise;
         },
-        getOptions: function () {
+        getOptions: function() {
           var deferred = $q.defer();
-          var deferredDocumentType = $q.defer();
+
+          $q.all({
```

Comment:
I tried to implement your solution, but doing so, the promise does not get resolved.